src/libraries/Common/src/Interop/OSX/System.Security.Cryptography.Native.Apple/Interop.X509.cs

```diff
@@ -14,33 +14,41 @@ internal static partial class Interop
 {
     internal static partial class AppleCrypto
     {
-        [DllImport(Libraries.AppleCryptoNative)]
-        private static extern int AppleCryptoNative_X509GetRawData(
+        [GeneratedDllImport(Libraries.AppleCryptoNative)]
+        private static partial int AppleCryptoNative_X509GetRawData(
             SafeSecCertificateHandle cert,
             out SafeCFDataHandle cfDataOut,
             out int pOSStatus);
 
-        [DllImport(Libraries.AppleCryptoNative)]
-        private static extern int AppleCryptoNative_X509GetPublicKey(SafeSecCertificateHandle cert, out SafeSecKeyRefHandle publicKey, out int pOSStatus);
+        [GeneratedDllImport(Libraries.AppleCryptoNative)]
+        private static partial int AppleCryptoNative_X509GetPublicKey(SafeSecCertificateHandle cert, out SafeSecKeyRefHandle publicKey, out int pOSStatus);
 
         internal static X509ContentType X509GetContentType(ReadOnlySpan<byte> data)
-            => X509GetContentType(ref MemoryMarshal.GetReference(data), data.Length);
+        {
+            unsafe
+            {
+                fixed (byte* dataPtr = &MemoryMarshal.GetReference(data))
+                {
+                    return X509GetContentType(dataPtr, data.Length);
+                }
+            }
+        }
 
         [DllImport(Libraries.AppleCryptoNative, EntryPoint = "AppleCryptoNative_X509GetContentType")]
-        private static extern X509ContentType X509GetContentType(ref byte pbData, int cbData);
+        private static unsafe extern X509ContentType X509GetContentType(byte* pbData, int cbData);
```

Comment:
Was this change done for perf, or does the source generator not yet support refs?

Reply:
This was done for perf reasons. It's better perf to use pointers and have no stub than to use refs and generate a stub that only pins a few parameters before the call.